### PR TITLE
Adding one section of error explanations

### PR DESCRIPTION
**Pull request by davidcok**

### CODE REVIEW

**Review by MikaelMayer — COMMENTED** [review]
Oh wow this is really useful.
A few changes to make this PR perfect

File: docs/HowToFAQ/ERROR_DataTypeName.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+title: "Error: name of datatype (String) is used as a function"?
+---
+
+How do I fix this error message: "name of datatype (String) is used as a function"?
+
+```dafny
```
[the remark this answers is not in the document]
Comment (davidcok) [reply docs/HowToFAQ/ERROR_DataTypeName.md:7]: I will make a separate task to move all examples (that are not just snippets) into separate files, as it applies to more than just these FAQ/Error entries.

File: docs/HowToFAQ/ERROR_NoLHS.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+title: "Error: cannot establish the existence of LHS values that satisfy the such-that predicate"
+---
+
+Here is example code that produces this error:
+```dafny
+{% include_relative ERROR_NoLHS.dfy %}
+```
+
+
+When a _such-that_ (`:|`) initialization is used, Dafny must be able to establish that there is at least one value
```
[the remark this answers is not in the document]
Comment (davidcok) [reply docs/HowToFAQ/ERROR_NoLHS.md:11]: Fixed

File: docs/HowToFAQ/ERROR_NoLHS.md [hunk 3]
```diff
@@ -9,7 +9,10 @@ Here is example code that produces this error:
 
 
 When a _such-that_ (`:|`) initialization is used, Dafny must be able to establish that there is at least one value
-that satisfies the predicate given on the RHS. In this case if the map `m` is empty, there is no such value,
+that satisfies the predicate given on the RHS. 
+That is, in this case, it must prove
+`assert exists k | k in m;`.
```
Comment (MikaelMayer) [inline docs/HowToFAQ/ERROR_NoLHS.md:14]:
```suggestion
`assert exists k :: k in m;`.
```
Sorry, my bad, I just checked, and the single `|` syntax is invalid. It should but it's not.

**Review by MikaelMayer — APPROVED** [review]

**Pull request merged**